core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -140,7 +145,7 @@ class BrokerMetadataPublisher(conf: KafkaConfig,
       // Apply topic deltas.
       Option(delta.topicsDelta()).foreach { topicsDelta =>
         // Notify the replica manager about changes to topics.
-        replicaManager.applyDelta(topicsDelta, newImage)
+        replicaManager.applyDelta(topicsDelta, newImage, versionDelta)
```

Comment:
If we like this approach, we would need to pass the version delta to the other components that get updated here.